### PR TITLE
Feature/67 sorting newly added wallets

### LINKED ISSUE #67: Sorting newly added wallets
After adding several wallets, they are displayed in the wrong order.
To illustrate this issue, add wallets with the following names and in the following order:
- a
- b
- c
After refreshing the page, the wallets are displayed sorted by name.

### PR DESCRIPTION
**Pull request by DawidStuzynski**

### CODE REVIEW

File: src/app/domains/wallets/domains.wallets.gateway.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export class DomainsWalletsGateway {
   }
 
   public updateWallet(id: number, { name }: IWalletPayload): Observable<IWalletApiResponse> {
-    return this.http.put<IWalletApiResponse>(API_WALLETS_URL, { id, name });
+    return this.http.patch<IWalletApiResponse>(API_WALLETS_URL + `/${id}`, { name });
```
Comment (purbanski-deftcode) [inline src/app/domains/wallets/domains.wallets.gateway.ts]: Should be ``${API_WALLETS_URL}/${id}``

File: src/app/pages/wallets/management/pages-wallets-management.component.ts [hunk 2]
```diff
@@ -57,47 +57,50 @@ export class PagesWalletsManagementComponent implements OnInit {
...
     this.myWalletsService.createWallet(WalletsManagementItem.create({ name })).subscribe({
-        next: (wallet: WalletsManagementItem) => {
-          this.myWalletsData.data = [wallet, ...this.myWalletsData.data!];
-          this.systemNotificationsService.showNotification({ message: 'Congratulations! Your wallet was created successfully.' });
-        },
-        error: () => {
-          this.systemNotificationsService.showNotification({ message: 'Sorry. Something went wrong and your wallet was not saved. Contact administrator.' });
-        },
+      next: (wallet: WalletsManagementItem) => {
+        this.myWalletsData.data = [wallet, ...this.myWalletsData.data!];
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/management/pages-wallets-management.component.ts]: You can just go with:
```typescript
this.myWalletsData.data = [wallet, ...this.myWalletsData.data!].sort(here_rest_of_the_code);
```

File: src/app/pages/wallets/management/pages-wallets-management.component.ts [hunk 3]
```diff
@@ -57,47 +57,50 @@ export class PagesWalletsManagementComponent implements OnInit {
...
-                    return updatedWallet;
-                }
-                return walletItem;
-            });
-        },
-        error: () => {
-            this.systemNotificationsService.showNotification({ message: 'Some server error during updating' });
-        },
+      next: (updatedWallet: WalletsManagementItem) => {
+        this.myWalletsData.data = this.myWalletsData.data!.map(walletItem => {
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/management/pages-wallets-management.component.ts:75]: Seems like we should also sort wallets when updating a wallet. If we rename a wallet the old position can be wrong and need to be adjusted.

**Review by purbanski-deftcode — CHANGES_REQUESTED** [review]

**Comment by purbanski-deftcode** [comment]
Closes #67

**Review by purbanski-deftcode — APPROVED** [review]

**Review by mirekgab — APPROVED** [review]

**Pull request merged**